### PR TITLE
feat: Create ksql cluster component module

### PR DESCRIPTION
**Pull request by jai**
Please confirm that you have done the following before requesting reviews:

- [x] I have confirmed that the PR type is appropriate for the change I am making according to the [Honest Pull Request and Commit Message Naming Conventions](https://www.notion.so/honestbank/Pull-Request-and-Commit-Message-Naming-Conventions-bd97f2cbb34c4c73b1ff3a3e384b850c).
- [x] I have typed an adequate description that explains **why** I am making this change.
- [x] I have installed and run standard pre-commit hooks that lints and validates my code.

### Description

Create a ksqlDB cluster component module for use by [kp](https://github.com/honestbank/kp).

- - -
This change is [Reviewable](https://reviewable.io/reviews/honestbank/terraform-confluent-kafka/8)

### CODE REVIEW

**Comment by jai** [comment]
Current dependencies on/for this PR:
* main
  * **PR #8** <a href="https://app.graphite.dev/github/pr/honestbank/terraform-confluent-kafka/8" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/honestbank/terraform-confluent-kafka/8?utm_source=stack-comment).

File: modules/ksql-cluster/ksql-cluster.tf [hunk 1]
```diff
@@ -0,0 +1,13 @@
+resource "confluent_ksql_cluster" "ksqldb_cluster" {
+  display_name = "example"
```
Comment (cyberhck) [inline modules/ksql-cluster/ksql-cluster.tf]: are we going to use `example` as the display name? What does that mean? Is it the name of the cluster? Should this be parameterized?

**Review by bludot — APPROVED** [review]

File: examples/create-env-cluster-topics/outputs.tf [hunk 2]
```diff
@@ -1,3 +1,16 @@
+output "bigquery_connector_id" {
+  value = module.honest_labs_connector_bigquery_sink.0.connector_id
```
Comment (poom) [inline examples/create-env-cluster-topics/outputs.tf]: Should it be fail if count = 0?

File: modules/ksql-cluster/ksql-cluster.tf (hunk 1 above)
Comment (jai) [reply modules/ksql-cluster/ksql-cluster.tf]: fixed my bad...been a long day 😂

**Review by dantodor — APPROVED** [review]

**Review by poom — APPROVED** [review]

**Comment by poom** [comment]
but test failed!

File: examples/create-env-cluster-topics/outputs.tf (hunk 2 above)
Comment (jai) [reply examples/create-env-cluster-topics/outputs.tf]: fixed/updated

**Comment by jai** [comment]
> but test failed!

finally passed! :)

**Pull request merged**